File: scripts/analyze-who-data.js
#!/usr/bin/env node
/**
 * 分析WHO data.xlsx文件，提取PPE相关数据
 */

const xlsx = require('xlsx');
const fs = require('fs');

const FILE_PATH = '/Users/maxiaoha/Desktop/mdlooker/mdlooker/WHO data.xlsx';

async function main() {
  console.log('=== WHO数据文件分析 ===\n');

  // 检查文件是否存在
  if (!fs.existsSync(FILE_PATH)) {
    console.error('错误: 文件不存在:', FILE_PATH);
    return;
  }

  const stats = fs.statSync(FILE_PATH);
  console.log(`文件大小: ${(stats.size / 1024 / 1024).toFixed(2)} MB\n`);

  // 读取Excel文件
  console.log('正在读取Excel文件...');
  const workbook = xlsx.readFile(FILE_PATH);

  console.log('\n=== 工作表列表 ===');
  workbook.SheetNames.forEach((name, i) => {
    console.log(`  ${i + 1}. ${name}`);
  });

  // 分析每个工作表
  console.log('\n=== 工作表详细分析 ===');

  for (const sheetName of workbook.SheetNames) {
    console.log(`\n--- 工作表: "${sheetName}" ---`);

    const worksheet = workbook.Sheets[sheetName];
    const data = xlsx.utils.sheet_to_json(worksheet, { header: 1 });

    console.log(`  行数: ${data.length}`);

    if (data.length > 0) {
      // 显示表头
      const headers = data[0];
      console.log(`  列数: ${headers.length}`);
      console.log('  表头:');
      headers.forEach((h, i) => {
        console.log(`    ${i + 1}. ${h}`);
      });

      // 显示前3行数据示例
      if (data.length > 1) {
        console.log('  数据示例（前3行）:');
        for (let i = 1; i < Math.min(4, data.length); i++) {
          console.log(`    行${i}:`, data[i].slice(0, 5).join(' | '));
        }
      }
    }
  }

  // 查找PPE相关内容
  console.log('\n\n=== PPE相关内容搜索 ===');

  const ppeKeywords = ['PPE', 'mask', '口罩', 'respirator', 'glove', '手套', 'gown', '防护服', 'protection', '防护'];

  for (const sheetName of workbook.SheetNames) {
    const worksheet = workbook.Sheets[sheetName];
    const data = xlsx.utils.sheet_to_json(worksheet, { header: 1 });

    if (data.length === 0) continue;

    const headers = data[0];
    const ppeColumns = [];

    // 查找包含PPE关键词的列
    headers.forEach((h, i) => {
      const headerStr = String(h).toLowerCase();
      if (ppeKeywords.some(kw => headerStr.includes(kw.toLowerCase()))) {
        ppeColumns.push({ index: i, header: h });
      }
    });

    if (ppeColumns.length > 0) {
      console.log(`\n工作表 "${sheetName}" 发现PPE相关列:`);
      ppeColumns.forEach(col => {
        console.log(`  - ${col.header}`);
      });
    }
  }

  console.log('\n\n分析完成!');
}

main().catch(console.error);

File: scripts/analyze_who_data.py
#!/usr/bin/env python3
"""
分析WHO data.xlsx文件，提取PPE相关数据
"""

import pandas as pd
import sys

FILE_PATH = '/Users/maxiaoha/Desktop/mdlooker/mdlooker/WHO data.xlsx'

def main():
    print("=== WHO数据文件分析 ===\n")
    
    # 获取文件信息
    import os
    file_size = os.path.getsize(FILE_PATH)
    print(f"文件大小: {file_size / 1024 / 1024:.2f} MB\n")
    
    # 读取Excel文件的所有sheet名称
    print("正在读取工作表列表...")
    xl = pd.ExcelFile(FILE_PATH)
    sheet_names = xl.sheet_names
    
    print(f"\n=== 工作表列表 ({len(sheet_names)}个) ===")
    for i, name in enumerate(sheet_names, 1):
        print(f"  {i}. {name}")
    
    # 分析每个工作表
    print("\n=== 工作表详细分析 ===")
    
    ppe_keywords = ['ppe', 'mask', 'respirator', 'glove', 'gown', 'protection', 
                    'surgical', 'medical', 'n95', 'ffp', 'kn95']
    
    for sheet_name in sheet_names[:5]:  # 只分析前5个工作表
        print(f"\n--- 工作表: '{sheet_name}' ---")
        
        try:
            # 只读取前5行来分析结构
            df = pd.read_excel(FILE_PATH, sheet_name=sheet_name, nrows=5)
            
            rows = len(pd.read_excel(FILE_PATH, sheet_name=sheet_name, usecols=[0]))
            cols = len(df.columns)
            
            print(f"  行数: {rows}, 列数: {cols}")
            print(f"  表头（前10列）:")
            
            for i, col in enumerate(df.columns[:10], 1):
                print(f"    {i}. {col}")
            
            if cols > 10:
                print(f"    ... 还有 {cols - 10} 列")
            
            # 显示前3行数据
            print(f"  数据示例（前3行，前5列）:")
            for idx, row in df.head(3).iterrows():
                values = [str(v)[:30] for v in row.values[:5]]
                print(f"    行{idx+1}: {' | '.join(values)}")
            
            # 查找PPE相关列
            ppe_cols = []
            for col in df.columns:
                col_str = str(col).lower()
                if any(kw in col_str for kw in ppe_keywords):
                    ppe_cols.append(col)
            
            if ppe_cols:
                print(f"  发现PPE相关列:")
                for col in ppe_cols:
                    print(f"    - {col}")
                    
        except Exception as e:
            print(f"  错误: {e}")
    
    print("\n\n分析完成!")

if __name__ == '__main__':
    main()
